packages/fluentui/react-northstar/src/components/SplitButton/SplitButton.tsx

```diff
@@ -87,149 +103,131 @@ export interface SplitButtonProps
   toggleButton?: ShorthandValue<SplitButtonToggleProps>;
 }
 
-export interface SplitButtonState {
-  isFromKeyboard: boolean;
-  open: boolean;
-}
-
 export const splitButtonClassName = 'ui-splitbutton';
 
-class SplitButton extends AutoControlledComponent<WithAsProp<SplitButtonProps>, SplitButtonState> {
-  static create: ShorthandFactory<SplitButton>;
-
-  static displayName = 'SplitButton';
+export type SplitButtonStylesProps = Required<Pick<SplitButtonProps, 'size'>> & { isFromKeyboard: boolean };
+
+export const SplitButton: React.FC<WithAsProp<SplitButtonProps>> &
+  FluentComponentStaticProps<SplitButtonProps> & {
+    Toggle: typeof SplitButtonToggle;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(SplitButton.displayName, context.telemetry);
+  setStart();
+
+  const {
+    button,
+    disabled,
+    menu,
+    primary,
+    secondary,
+    toggleButton,
+    size,
+    position,
+    align,
+    flipBoundary,
+    overflowBoundary,
+    positionFixed,
+    offset,
+    unstable_pinned,
+    className,
+    design,
+    styles,
+    accessibility,
+    variables,
+  } = props;
+
+  const targetRef = React.useRef<HTMLDivElement>();
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(SplitButton.handledProps, props);
+  const [open, setOpen] = useAutoControlled({
+    defaultValue: props.defaultOpen,
+    value: props.open,
+    initialValue: false,
+  });
 
-  static Toggle = SplitButtonToggle;
+  const [fromKeyboard, setIsFromKeyboard] = useAutoControlled({
```

Comment:
```suggestion
  const [isFromKeyboard, setIsFromKeyboard] = React.useState<boolean>(false)
```

Reply:
If I change `fromKeyboard` it will conflict with `isFromKeyboard()` imported from `utils`

Reply:
Ah, I see, damn 😭

Reply:
In this case I would suggest to rename export for util:

```
import { isFromKeyboard as checkIsFromKeyboard } from '...'
```

Reply:
Ok,